### ISSUE TITLE
`go-yara` 4.3.3 breaks warning ID construction

### ISSUE THREAD

**Issue by egibs** [chainguard-dev/bincapz#384]
`go-yara` 4.3.3 added guardrails to prevent accessing `nil` Rule Warnings which breaks our warning ID string construction.

As of `4.3.3`, the only valid strings we can use now are `ycw.Rule` and `ycw.Text` which will not match the `r.Namespace()` and `r.Identifier()` checks in `id` later on.

**Comment by egibs** [comment]
Here's the commit that swapped from a `*Rule` type to `string` type for the Rule field:
https://github.com/hillu/go-yara/commit/f03acb3bf4ed7955549b22c1906e432189da7ba

**Comment by egibs** [comment]
Resolved in #386 -- the `CompilerWarning` `Rule` field is now a string and encodes the rule namespace and identifier in the same, period-delimited string. 

To retrieve the identifier that we can match against when iterating through the compiled rules, we need to do this now:
```go
identifier := strings.Split(ycw.Rule, ".")[len(strings.Split(ycw.Rule, "."))-1]
```
.
